MonstieWash/Assets/Scripts/Animation Control/MonsterAnimControl.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[Serializable]
+public class MoodToAnimName
+{
+    public string moodName;
+    public string animName;
+}
+
+public class MonsterAnimControl : MonoBehaviour
+{
+    private MonsterBrain m_mimicAI;   
+    private string m_currentMood;   // tracks the monster's current mood (accurate to last frame)
+    private Animator m_myAnimator;
+
+    [SerializeField] private List<MoodToAnimName> moodToAnimationMap; // maps the name of moods to their animation names
+
+    void Start()
+    {
+        m_myAnimator = GetComponent<Animator>();
+    }
+
+    private void OnEnable()
+    {
+        if (m_mimicAI == null)
+        {
+            m_mimicAI = FindFirstObjectByType<MonsterBrain>();
+        }
+    }
+
+    void Update()
+    {
+        UpdateAnimations();
+    }
+
+    /// <summary>
+    // Changes the monster's animation (using the Animator component) to fit its current mood
+    /// </summary>
+    void UpdateAnimations()
+    {
+        // Get the name of the mood with the highest float value from mimicAI MoodData
+        string highestMood = m_mimicAI.GetHighestMood();
+        // Match that name to the name of that mood's animation
+        MoodToAnimName animToPlay = moodToAnimationMap.Find(x => x.moodName == highestMood);
```

Comment:
Don't forget to use var, and also we'll need to make some adjustments since `MoodToAnimation` has been changed.
```suggestion
        // Get the name of the mood with the highest float value from mimicAI MoodData
        var highestMood = m_mimicAI.GetHighestMood();
        // Match that name to the name of that mood's animation
        var animToPlay = moodToAnimationMap.Find(obj => obj.mood.MoodName == highestMood).animation;
```